### PR TITLE
MoleculeNet Cheatsheet

### LINKED ISSUE #3330: Add MoleculeNet Cheatsheet
We should have a MoleculeNet cheatsheet that provides a listing of all the available datasets in MoleculeNet like we have with the models cheatsheet

### PR DESCRIPTION
**Pull request by JoseAntonioSiguenza**
Cheatsheet of deepchem's datasets

Fix #3330 

This PR adds a cheatsheet on MoleculeNet datasets for new users to identify more easily the type of data each set contains. 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by JoseAntonioSiguenza** [comment]
Screen Shot 2023-04-25 at 12 45 11
Screen Shot 2023-04-25 at 12 45 49

**Comment by JoseAntonioSiguenza** [comment]
@rbharath I wasn't able to run `make html` in `deepchem/docs directory` (I think it's an update problem of my branch). Besides, I ran it locally and here are the renders. However, I think the table is too large and could be shocking for a user. I'm happy know commentaries.

**Comment by JoseAntonioSiguenza** [comment]
Here is a render from `doc` directory.
Screen Shot 2023-04-28 at 08 29 51
Screen Shot 2023-04-28 at 08 30 07
Screen Shot 2023-04-28 at 08 30 18

File: docs/source/api_reference/moleculenet_datasets_desccription.csv [hunk 1]
```diff
@@ -0,0 +1,42 @@
+﻿Name,Description,Type ,Data Points,Reference
```
Comment (rbharath) [inline docs/source/api_reference/moleculenet_datasets_desccription.csv:1]: "desccription.csv" -> "description.csv"

File: docs/source/api_reference/moleculenet_cheatsheet.rst [hunk 2]
```diff
@@ -0,0 +1,18 @@
+MoleculeNet Cheatsheet
```
Comment (rbharath) [inline docs/source/api_reference/moleculenet_cheatsheet.rst:1]: You need to embed this in moleculenet.rst

**Review by rbharath — COMMENTED** [review]
Left a few comments. Once addressed and rebased good to merge

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**